src/main/java/com/openlattice/shuttle/dates/DecadeChangeHelper.java

```diff
@@ -0,0 +1,56 @@
+package com.openlattice.shuttle.dates;
+
+import com.openlattice.shuttle.util.Cached;
+import com.openlattice.shuttle.util.Constants;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+
+public class DecadeChangeHelper {
+    /**
+     *
+     * Dealing with the fact that a two-year year-pattern should not all be in 20YY
+     *
+     */
+    public static boolean checkDatePatternIsTwoDigitYear( String datePattern ) {
+        boolean yyMatch = Cached.getMatcherForString( datePattern, ".*yy.*" ).matches();
+        boolean yyyyMatch = Cached.getMatcherForString( datePattern, ".*yyyy.*" ).matches();
+        return yyMatch && !yyyyMatch;
+    }
+
+    public static LocalDateTime fixTwoYearPatternLocalDate (
+            LocalDateTime parsedDateTime,
+            String datePattern
+    ) {
+        if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
+            if ( ( parsedDateTime.getYear() - LocalDate.now().getYear() ) > Constants.DECADE_CUTOFF ) {
+                parsedDateTime = parsedDateTime.withYear( parsedDateTime.getYear() - 100 );
+            }
+        }
+        return parsedDateTime;
+    }
+
+    public static LocalDate fixTwoYearPatternLocalDate (
```

Comment:
Isn't this the same function as above ^ ?

Reply:
Yeah, the type of input is different.  Or can I put those together in a way?

Reply:
You could do something like this:
```
var localDate = fixTwoYearPatternLocalDate( parsedDateTime.toLocalDate(), datePattern );
return LocalDateTime.of( localDate, parsedDateTime.toLocalTime() );
```

Reply:
Wouldn't that strip the time from the localDateTime?
And inversely `parsedDate.toLocalDateTime()` doesn't exist...

Reply:
This is to replace the fix of `LocalDateTime` to re-use the `LocalDate` function, because you only need the date part (and not the time part) of the datetime to do the decade fix. So, the two functions would look like this:
```
    public static LocalDateTime fixTwoYearPatternLocalDateTime (
            LocalDateTime parsedDateTime,
            String datePattern
    ) {
        var localDate = fixTwoYearPatternLocalDate( parsedDateTime.toLocalDate(), datePattern );
        return LocalDateTime.of( localDate, parsedDateTime.toLocalTime() );
    }

    public static LocalDate fixTwoYearPatternLocalDate (
            LocalDate parsedDate,
            String datePattern
    ) {
        if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
            if ( ( parsedDate.getYear() - LocalDate.now().getYear() ) > Constants.DECADE_CUTOFF ) {
                parsedDate = parsedDate.withYear( parsedDate.getYear() - 100 );
            }
        }
        return parsedDate;
    }
```

Reply:
also bumping this suggestion (related) https://github.com/openlattice/shuttle/pull/141#discussion_r378466780